source/runtime.cpp

```diff
@@ -607,6 +607,10 @@ void reshade::runtime::on_present()
 		}
 	}
 
+#if RESHADE_ADDON
+	invoke_addon_event<addon_event::reshade_available_back_buffer>(this);
```

Comment:
As far as I can tell this is basically equivalent to `addon_event::present`, except when the back buffer is multisampled. But in that case this wouldn't be useable too since `effect_runtime::get_current_back_buffer()` returns the real back buffer and not the resolved one.

Reply:
How to capture "-original" image?
In develop, this event name was `reshade_original`

Reply:
You can use this little hack to capture a screenshot in the `present` event (which would ve the original image before effects) despite not technically having access to the effect runtime (works because the swapchain pointer indirectly is also an effect runtime still):
```c++
static void on_present(reshade::api::swapchain *swapchain, ...)
{
    const auto runtime = static_cast<reshade::api::effect_runtime *>(swapchain);
    runtime->capture_screenshot(...);
}
```

Reply:
removed. [Add-on side](https://github.com/seri14/reshade-addons/commit/d8f16709fcd4ba820b2f2d79d330b9769dd7a4b3)